lib/v2/screens/authentication/recover/recover_account_found/recover_account_found_screen.dart

```diff
@@ -23,67 +26,86 @@ class RecoverAccountFoundScreen extends StatelessWidget {
-                          Icons.copy,
-                          color: AppColors.white,
-                        ),
-                        onPressed: () {
-                          Clipboard.setData(ClipboardData(text: state.linkToActivateGuardians)).then(
-                            (value) {
-                              SnackBarInfo("Copied", ScaffoldMessenger.of(context)).show();
-                            },
-                          );
-                        },
-                      ),
-                      controller: TextEditingController(text: state.linkToActivateGuardians),
-                    ),
-                  ),
-                  Padding(
-                    padding: const EdgeInsets.all(16),
-                    child: Row(
-                      children: [
-                        Text("TODO", style: Theme.of(context).textTheme.button1),
-                        Text("/TODO", style: Theme.of(context).textTheme.button1),
-                        const SizedBox(
-                          width: 24,
-                        ),
-                        Flexible(
-                          child: Text(
-                            "Guardians have accepted your request to recover your account",
-                            style: Theme.of(context).textTheme.buttonLowEmphasis,
+            switch (state.pageState) {
+              case PageState.initial:
+                return const SizedBox.shrink();
+              case PageState.loading:
+                return const FullPageLoadingIndicator();
+              case PageState.failure:
+                return FullPageErrorIndicator(errorMessage: state.errorMessage);
+              case PageState.success:
+                switch (state.recoveryStatus) {
+                  case RecoveryStatus.WAITING_FOR_GUARDIANS_TO_SIGN:
+                    return Padding(
+                      padding: const EdgeInsets.all(8.0),
+                      child: Column(
+                        children: [
+                          Padding(
+                            padding: const EdgeInsets.only(left: 8, right: 8, top: 8),
+                            child: TextFormFieldCustom(
+                              enabled: false,
+                              labelText: 'Link to Activate Key Guardians',
+                              suffixIcon: IconButton(
+                                icon: const Icon(
+                                  Icons.copy,
+                                  color: AppColors.white,
+                                ),
+                                onPressed: () {
+                                  Clipboard.setData(ClipboardData(text: state.linkToActivateGuardians)).then(
+                                    (value) {
+                                      SnackBarInfo("Copied", ScaffoldMessenger.of(context)).show();
+                                    },
+                                  );
+                                },
```

Comment:
I thought we agreed not to dialogs / snacks in the onPress only bloc events.

Reply:
true, this was a copy/paste. 
will fix in next PR, left a todo